File: src/test/java/at/fhhagenberg/sqe/di/ElevatorControlSystemProvider.java
package at.fhhagenberg.sqe.di;

import at.fhhagenberg.sqe.entity.*;
import com.google.inject.Provider;
import java.util.ArrayList;
import java.util.List;

public class ElevatorControlSystemProvider implements Provider<ElevatorControlSystem> {

    public static final int FLOORS_COUNT = 5;
    public static final int ELEVATORS_COUNT = 3;
    public static final int FEET_PER_FLOOR = 14;
    public static final int CAPACITY_PER_ELEVATOR = 8;

    @Override
    public ElevatorControlSystem get() {
        return new ElevatorControlSystem(
                0L,
                FEET_PER_FLOOR * FLOORS_COUNT,
                createMockBuildingFloors(FLOORS_COUNT),
                createMockElevators(ELEVATORS_COUNT, FLOORS_COUNT)
        );
    }

    private List<BuildingFloor> createMockBuildingFloors(int totalNumberOfFloors) {
        List<BuildingFloor> result = new ArrayList<>(totalNumberOfFloors);
        for (int floorNumber = 0; floorNumber < totalNumberOfFloors; floorNumber++) {
            result.add(new BuildingFloor(
                    floorNumber,
                    false,
                    false));
        }
        return result;
    }

    private List<Elevator> createMockElevators(int totalNumberOfElevators, int totalNumberOfFloors) {
        List<Elevator> result = new ArrayList<>(totalNumberOfElevators);
        for (int elevatorNumber = 0; elevatorNumber < totalNumberOfElevators; elevatorNumber++) {
            result.add(new Elevator.Builder()
                    .setElevatorNumber(elevatorNumber)
                    .setDoorState(DoorState.CLOSED)
                    .setAcceleration(0)
                    .setButtons(createMockButtons(elevatorNumber, totalNumberOfFloors))
                    .setCapacity(CAPACITY_PER_ELEVATOR)
                    .setCommittedDirection(Direction.UNCOMMITTED)
                    .setCurrentFloor(0)
                    .setCurrentPosition(0)
                    .setCurrentSpeed(0)
                    .setCurrentWeight(0)
                    .setTargetFloor(0)
                    .setServicedFloors(createMockServicedFloors(elevatorNumber, totalNumberOfFloors))
                    .build()
            );
        }
        return result;
    }

    private List<FloorButton> createMockButtons(int elevatorNumber, int totalNumberOfFloors) {
        List<FloorButton> result = new ArrayList<>(totalNumberOfFloors);
        for (int floorNumber = 0; floorNumber < totalNumberOfFloors; floorNumber++) {
            result.add(new FloorButton(
                    elevatorNumber,
                    floorNumber,
                    false
            ));
        }
        return result;
    }

    private List<ServicedFloor> createMockServicedFloors(int elevatorNumber, int totalNumberOfFloors) {
        List<ServicedFloor> result = new ArrayList<>(totalNumberOfFloors);
        for (int floorNumber = 0; floorNumber < totalNumberOfFloors; floorNumber++) {
            result.add(new ServicedFloor(
                    elevatorNumber,
                    floorNumber,
                    true
            ));
        }
        return result;
    }
}

File: src/main/kotlin/at/fhhagenberg/sqe/di/provider/ElevatorControlSystemProvider.kt
package at.fhhagenberg.sqe.di.provider

import at.fhhagenberg.sqe.entity.*
import com.google.inject.Provider
import java.util.*

class ElevatorControlSystemProvider : Provider<ElevatorControlSystem> {

    private val totalNumberOfFloors = 6
    private val totalNumberOfElevators = 4
    private val feetPerFloor = 10
    private val capacityPerElevator = 8

    override fun get() = ElevatorControlSystem(
            100L,
            feetPerFloor,
            createMockBuildingFloors(),
            createMockElevators()
    )

    private fun createMockBuildingFloors(): List<BuildingFloor> {
        val result: MutableList<BuildingFloor> = ArrayList(totalNumberOfFloors)
        for (floorNumber in 0 until totalNumberOfFloors) {
            result.add(BuildingFloor(
                    floorNumber,
                    isDownActive = floorNumber == 2,
                    isUpActive = false))
        }
        return result
    }

    private fun createMockElevators(): List<Elevator> {
        val result: MutableList<Elevator> = ArrayList(totalNumberOfElevators)
        for (elevatorNumber in 0 until totalNumberOfElevators) {
            result.add(Elevator.Builder()
                    .elevatorNumber(elevatorNumber)
                    .doorState(DoorState.CLOSED)
                    .acceleration(1)
                    .buttons(createMockButtons(elevatorNumber))
                    .capacity(capacityPerElevator)
                    .committedDirection(if (elevatorNumber == 0) Direction.DOWN else if (elevatorNumber == 1) Direction.UP else Direction.UNCOMMITTED)
                    .currentFloor(if (elevatorNumber == 0) 5 else if (elevatorNumber == 1) 3 else 0)
                    .currentPosition(if (elevatorNumber == 0) 50 else if (elevatorNumber == 1) 30 else 0)
                    .currentSpeed(2)
                    .currentWeight(170)
                    .targetFloor(0)
                    .servicedFloors(createMockServicedFloors(elevatorNumber))
                    .build()
            )
        }
        return result
    }

    private fun createMockButtons(elevatorNumber: Int): List<FloorButton> {
        val result: MutableList<FloorButton> = ArrayList(totalNumberOfFloors)
        for (floorNumber in 0 until totalNumberOfFloors) {
            result.add(FloorButton(
                    elevatorNumber,
                    floorNumber,
                    (elevatorNumber == 1 || elevatorNumber == 3) && floorNumber == 1
            ))
        }
        return result
    }

    private fun createMockServicedFloors(elevatorNumber: Int): List<ServicedFloor> {
        val result: MutableList<ServicedFloor> = ArrayList(totalNumberOfFloors)
        for (floorNumber in 0 until totalNumberOfFloors) {
            result.add(ServicedFloor(
                    elevatorNumber,
                    floorNumber,
                    !(elevatorNumber == 1 && floorNumber == 2)
            ))
        }
        return result
    }
}